Client/Frontend/Widgets/AutocompleteTextField.swift

```diff
@@ -82,27 +87,51 @@ class AutocompleteTextField: UITextField, UITextFieldDelegate {
         }
     }
 
+    // `shouldChangeCharactersInRange` is called before the text changes, and SELtextDidChange is called after.
+    // Since the text has changed, remove the completion here, and SELtextDidChange will fire the callback to
+    // get the new autocompletion.
     func textField(textField: UITextField, shouldChangeCharactersInRange range: NSRange, replacementString string: String) -> Bool {
-        if completionActive && string.isEmpty {
-            // Characters are being deleted, so clear the autocompletion, but don't change the text.
-            removeCompletion()
-            return false
+        if completionActive {
+            if string.isEmpty {
+                // Characters are being deleted, so clear the autocompletion, but don't change the text.
+                removeCompletion()
+                return false
+            }
+            removeCompletionIfRequiredForEnteredString(string)
         }
-
         return true
     }
 
+    private func  removeCompletionIfRequiredForEnteredString(string: String) {
+        let actualEnteredString = text.substringToIndex(advance(text.startIndex, enteredTextLength, text.endIndex)) + string
+        // If actual entred text by user not starts with previous suggestion than remove completion.
```

Comment:
Nit: `If user-entered text does not start with previous suggestion then remove the completion.`